api/routes/measurements.js

```diff
@@ -96,7 +96,6 @@ module.exports = [
       if (request.query) {
```

Comment:
Ok, I think the last wrinkle I see here (famous last words!). You can see in the old webbooks.js, we generated the total count of all measurements based on what was being returned from the countries aggregation ([here](https://github.com/openaq/openaq-api/blob/develop/api/controllers/webhooks.js#L121)). Now we don't get that and we don't save it to Redis. There is some funky language to handle checking Redis for this value or querying the database to get the count [here](https://github.com/openaq/openaq-api/blob/develop/api/controllers/measurements.js#L35). Given the new approach, I think we need to save the total count to the database and grab it from there. At a minimum, we need to somehow have the total count easy to get (or else all `/measurements` calls will timeout). It'd be better if we could get things like `/measurements?country=CN`, etc also to have the counts pre-calculated (which I know we have in the database now), but that's a bonus.

Reply:
Thanks @vgeorge. I think the way you implemented the counts query will speed up a number of the requests, which is great! 

Please see http://localhost:3004/v1/measurements?value_from=10 as a failing example though. There are a number of query variables supported at https://docs.openaq.org/#api-Measurements that the `locations` table doesn't support. I think it may be best to check to see if the variables passed in the query are all supported by the `locations` table and if not, fall back to querying across the database. I think that code was already in the controller, so hopefully this is just an if/else.

Reply:
Hey, @jflasher, I've changed the controller, including tests covering the parameters you mentioned.

Now, by default, it will fetch counts from the measurements aggregations available at `locations` table. If `value_*/date_*` is passed, it will fetch from `measurements`. I know this incurs in returning smaller (and incorrect) counts for the second case, but I didn't saw a way of return valid counts for that case if the full history is unavailable. Currently API do not seems to be working when these parameters are used:

https://api.openaq.org/v1/measurements?parameter=co&value_from=100&value_from=1000

Reply:
Thanks @vgeorge, I know this is a bit weird, but I think this is the best we can do right now given how we have the data separated about between a database and S3. If we could just stop using these counts, it would actually make things a lot easier. I blame this on @scisco as I was just following his design on a previous project. :)

I'll try and take a look at this soon and I should hopefully be able to deploy to staging.

Reply:
Blame everything on me :)